### PR TITLE
SearchField: Initial implementation

### PR DESCRIPTION
**Pull request by Dor-Snapiri**
Stylized search field.

image

### CODE REVIEW

File: src/SearchField/SearchField.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { useState } from 'react'
+import './SearchField.scss'
+import MagnifyingGlassIcon from '../assets/magnifyingGlassIcon.svg?react'
+import CloseIcon from '../assets/closeIcon.svg?react'
+
+type SearchFieldProps = {
+    name: string
```
Comment (roy-guggenheim) [inline src/SearchField/SearchField.tsx:7]: Why "name"? This is a generic search component. Consider renaming it to simply "value" or something else more neutral.

File: src/SearchField/SearchField.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+export default function SearchField({ name, placeholder = "Search" }: SearchFieldProps) {
+    const [inputValue, setInputValue] = useState('');
+
+    const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setInputValue(event.target.value)
+    }
+
+    const handleClear = () => {
+        setInputValue('')
+    }
```
Comment (roy-guggenheim) [inline src/SearchField/SearchField.tsx:20]: Redundant declaration. Simply pass it in the `input` props.

BTW, this is a comment that might be considered as "nitpicking" but it's not really. If we're striving for excellence, which we are, then we always need to "think performance", especially in the FE implementations. Although this may seem like a small issue, every function declaration adds a small piece of code that increases the call stack and will need to be garbage-collected at some point in the life cycle of the program and each function also consumes memory for its own execution context even when not actively running.
On the other hand, sometimes it's worth it, e.g., a function that is not that simple and does several things, or a non-trivial operation - then it might deserve its own scope under a meaningful name.

File: src/SearchField/SearchField.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                <input
+                    type="text"
+                    placeholder={placeholder}
+                    name={`${name}Input`}
+                    value={inputValue}
+                    onChange={handleInputChange}
+                />
+            </div>
+
+            {inputValue && (
```
Comment (roy-guggenheim) [inline src/SearchField/SearchField.tsx:35]: Did you confirm the truth value of the empty string? 😉
Ways of doing that kind of tests:
1. In the browser console:
 
image

2. In node-console on your local terminal:
image

Comment (roy-guggenheim) [reply src/SearchField/SearchField.tsx:35]: And what about other edge cases? For instance, what if a string contains only white spaces? Consider using `trim()`.

**Review by roy-guggenheim — APPROVED** [review]
FSAM
